lib/p2p/node/supervisor.ex

```diff
@@ -91,14 +91,39 @@ defmodule Elixium.Node.Supervisor do
           body
           |> Jason.decode!()
           |> Enum.map(&peerstring_to_tuple/1)
-          |> Enum.filter(fn {_, port} -> port != nil end)
+          |> Enum.filter(fn {peer, port} -> port != nil && peer !== fetch_public_ip["ip"] && peer !== fetch_local_ip end)
 
         if peers == [], do: :not_found, else: peers
 
       {:error, _} -> :not_found
     end
   end
 
+
+  @doc """
+    On Connection, fetch our public ip and exclude it from the list of registered peers
```

Comment:
Since this *only* fetches our public IP, and doesn't do the other stuff in this @doc, lets just say that this function fetches our public IP

Reply:
Fixed